docs/root/configuration/http/http_filters/set_metadata_filter.rst

```diff
@@ -0,0 +1,52 @@
+.. _config_http_filters_set_metadata:
+
+Set Metadata
+============
+* :ref:`v3 API reference <envoy_v3_api_msg_extensions.filters.http.set_metadata.v3.Config>`
+* This filter should be configured with the name *envoy.filters.http.set_metadata*.
+
+This filters adds or updates dynamic metadata with static data.
+
+Dynamic metadata values are updated with the following scheme. If a key
```

Comment:
@envoyproxy/api-shepherds this differs from `MergeFrom` in that it supports merging structs together, e.g. if you have a disjoint namespace things compose as expected. I think this makes sense, but flagging this for consideration in case there is some alternate merging preferences.